reconcile/statuspage/page.py

```diff
@@ -124,13 +76,17 @@ class StatusPage(BaseModel):
     this desired state does. People can still manage components manually.
     """
 
-
-ProviderInitializer = Callable[
-    [StatusPageV1, str, ComponentBindingState], StatusPageProvider
-]
-
-_PROVIDERS: dict[str, ProviderInitializer] = {}
-
-
-def register_provider(provider: str, provider_init: ProviderInitializer) -> None:
-    _PROVIDERS[provider] = provider_init
+    @staticmethod
+    def init_from_page(
+        page: StatusPageV1,
+    ) -> "StatusPage":
+        """
+        Translate a desired state status page into a status page object.
+        """
+        return StatusPage(
```

Comment:
Same here can use classmethod and `Self`

Reply:
https://github.com/app-sre/qontract-reconcile/pull/4343/commits/7549eabf004bae065958b41195a124820ee1ea8e